### ISSUE TITLE
extra.pandas.column errorr when providing elements and dtype arguments

### ISSUE THREAD

**Issue by h-t-w** [HypothesisWorks/hypothesis#3299]
Hi all.

I am currently generating a dataframe strategy for my tests. One of the columns of the resulting pandas dataframe should be of dtype `int`. 

If I just specify the elements argument in the `column` call I sometimes get an `object` dtype instead of `int`. I personally find this irritating, but I can see that this might to be expected. But in the end, this is not what I want.

```python
from hypothesis import strategies as st
from hypothesis.extra import pandas as st_pd
import numpy as np

data = st_pd.data_frames(
        index=st_pd.range_indexes(min_size=5, max_size=50),
        columns=[
            st_pd.column("Volume", st.integers())
        ])
```
 
If I try to specify the dtype in the call (in addition to elements) to force the column dtype, the strategy fails and I get an error 'OverflowError: int too big to convert()'. This feels odd.

```python
data = st_pd.data_frames(
        index=st_pd.range_indexes(min_size=5, max_size=50),
        columns=[
            st_pd.column("Volume", st.integers(), dtype=np.int64)
        ]) 
```

If I just specify the dtype without the elements, it works, but I cannot set minimum and maximum values for the integers contained in the column this way.

```python
data = st_pd.data_frames(
        index=st_pd.range_indexes(min_size=5, max_size=50),
        columns=[
            st_pd.column("Volume", dtype=np.int64)
        ]) 
```

In short, the middle version feels like the one that I want, but it throws an error. So either I am doing something wrong - please let me know - or there is something off.

python 3.8
pandas 1.4.1
hypothesis 6.39.6

Best htw

**Comment by Zac-HD** [comment]
Ah, good question!  In the first case, your dtype is sometimes `object` because the `st.integers()` strategy can generate numbers too large to represent using a 64-bit integer.  Conversely, if you pass an unrestricted `st.integers()` strategy with `np.int64` as the dtype, then you get an `OverflowError` instead in those cases.

So I think what you want is to pass e.g. `st_pd.column("Volume", st.integers(-100_000, 100_000_000))`.

**Comment by h-t-w** [comment]
@Zac-HD 

Thanks for the clear explanation and the suggestion. I understand the issue now better. Your example will work for my needs.
